op-challenger/game/fault/trace/prestates/multi.go

```diff
@@ -75,8 +76,13 @@ func (m *MultiPrestateProvider) fetchPrestate(hash common.Hash, fileType string,
 	if resp.StatusCode != http.StatusOK {
 		return fmt.Errorf("%w from url %v: status %v", ErrPrestateUnavailable, prestateUrl, resp.StatusCode)
 	}
-	tmpFile := dest + ".tmp" + fileType // Preserve the file type extension so compression is applied correctly
-	out, err := ioutil.NewAtomicWriterCompressed(tmpFile, 0o644)
+	tmpFile := dest + ".tmp" + fileType // Preserve the file type extension so state decoding is applied correctly
+	var out *ioutil.AtomicWriter
+	if strings.HasSuffix(fileType, ".gz") {
+		out, err = ioutil.NewAtomicWriter(tmpFile, 0o644)
```

Comment:
It does seem like we should just use the AtomicWriter for all cases here since we never want to recompress anything?

Reply:
The challenger currently compresses singlethreaded Cannon prestates as they're provided as an uncompressed JSON. Though the disk saving from the compression isn't much. 23 MB of a singlethreaded prestate compresses to roughly 18 MB. Not worth it imo.
I'll tweak this to avoid compressing prestates.

Reply:
We can move to providing compressed JSON states if we want to now too.  Challenger checks for `.bin.gz` then `.json.gz` then `.json`.  But currently only the last one actually works - the other two get double compressed.  And I think the last one doesn't actually wind up compressed because we store it in a `.json` file anyway.

Reply:
I see, you're right. I was reading the compression for JSON prestates wrong.
Turns out that the [cannon state loader](https://github.com/ethereum-optimism/optimism/blob/develop/cannon/mipsevm/versions/state.go#L28) figures [out](https://github.com/ethereum-optimism/optimism/blob/eba495d2bc2db72e69cf3a93d8e5b7d6991bac66/op-service/jsonutil/json.go#L99C1-L122C2) whether to treat the prestate as compressed or not based on the file extension. It will be simpler for the op-challenger to preserve the prestate in the same format it fetched from the prestate server. We can let cannon figure out how to decode it.